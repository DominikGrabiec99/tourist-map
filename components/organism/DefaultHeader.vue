<template>
  <MoleculeHeader>
    <template #left>
      <MoleculeLogo />
    </template>
    <template #right>
      <div class="flex items-end">
        <nav class="flex items-end mr-4">
          <AtomLink
            v-for="option in MENU_OPTIONS"
            :key="option.id"
            :href="option.link"
            class="cursor-pointer text-lg text-sky-950 hover:text-sky-800 dark:text-sky-600 dark:hover:text-sky-500 mr-6 transition-color"
          >
            {{ option.text }}
          </AtomLink>
        </nav>
        <MoleculeMenuUser
          :is-user-menu-open="isUserMenuOpen"
          :menu-options="USER_MENU_LOGOUT_OPTIONS"
          @toggle-user-menu="toggleUserMenu"
        />
        <MoleculeLightModeIcon />
      </div>
    </template>
  </MoleculeHeader>
</template>

<script setup lang="ts">
import {
  USER_MENU_LOGOUT_OPTIONS,
  MENU_OPTIONS,
} from '@/constants/menuOptions';
const isUserMenuOpen = ref(false);

const toggleUserMenu = (value: boolean) => {
  isUserMenuOpen.value = value;
};

const clickBodyEvent = (event: MouseEvent) => {
  //@ts-ignore
  if (event.target?.id === 'user-menu') return;
  toggleUserMenu(false);
};

if (process.client) {
  document.body.addEventListener(
    'click',
    clickBodyEvent,
    true
  );
}
</script>

<style lang="scss">
.menu-enter-active,
.menu-leave-active {
  transition: opacity 0.5s ease;
}

.menu-enter-from,
.menu-leave-to {
  opacity: 0;
}
</style>
