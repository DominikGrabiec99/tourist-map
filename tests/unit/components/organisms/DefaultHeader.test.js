import { describe, it } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import DefaultHeader from '@/components/organism/DefaultHeader.vue';

describe('DefaultHeader.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(DefaultHeader);
  });

  it('render component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('toggle isUserMenuOpen', () => {
    expect(wrapper.vm.isUserMenuOpen).toBeFalsy();
    wrapper.vm.toggleUserMenu(true);
    expect(wrapper.vm.isUserMenuOpen).toBeTruthy();
  });

  it('handle clickBodyEvent', () => {
    wrapper.vm.toggleUserMenu(true);
    wrapper.vm.clickBodyEvent({
      target: {
        id: 'user-menu',
      },
    });
    expect(wrapper.vm.isUserMenuOpen).toBeTruthy();
    wrapper.vm.clickBodyEvent({
      target: {
        id: 'test-id',
      },
    });
    expect(wrapper.vm.isUserMenuOpen).toBeFalsy();
  });
});
