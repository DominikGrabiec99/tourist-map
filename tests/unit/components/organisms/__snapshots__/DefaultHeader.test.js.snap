// Vitest Snapshot v1

exports[`DefaultHeader.vue > render component 1`] = `<moleculeheader />`;
